### PR TITLE
Refs [TASK][35594] Handle event Favorite Books Screen

### PR DESCRIPTION
**Pull request by zGararz**
## Related Tickets
- [#Task 35594: Handle event Favorite Books Screen](https://edu-redmine.sun-asterisk.vn/issues/35594)
## WHAT
- 
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/itbook/ui/favorite/FavoriteBooksFragment.kt [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+    override fun showError(error: Exception?) {
+        Toast.makeText(activity, error.toString(), Toast.LENGTH_LONG).show()
+    }
+
+    override fun showLoading(isShow: Boolean) {
+        loadingDialogFragment?.let {
+            if (isShow) {
+                activity?.let { it1 -> it.show(it1.supportFragmentManager, "") }
```
Comment (thonc-2135) [inline app/src/main/java/com/example/itbook/ui/favorite/FavoriteBooksFragment.kt]: it1 đổi thành tên khác đi

File: app/src/main/java/com/example/itbook/ui/favorite/FavoriteBooksFragment.kt [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            } else {
+                it.dismiss()
+            }
+        }
+    }
+
+    private fun onBookClick(position: Int) {
+        val fragment = DetailBookFragment()
+        DetailBookFragment.apply {
```
Comment (thonc-2135) [inline app/src/main/java/com/example/itbook/ui/favorite/FavoriteBooksFragment.kt]: sao k applly từ trên mà còn thêm dòng lm j
```suggestion
        val fragment = DetailBookFragment().apply {
```

File: app/src/main/java/com/example/itbook/ui/favorite/FavoriteBooksFragment.kt [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            if (search in book.title.toLowerCase() || search in book.subtitle.toLowerCase()) {
+                filteredBooks.add(book)
+            }
+        }
+
+        return filteredBooks
+    }
+
+    private val textWatcher = object : TextWatcher {
+        override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
```
Comment (thonc-2135) [inline app/src/main/java/com/example/itbook/ui/favorite/FavoriteBooksFragment.kt]: biến mà lại để cuối thế

File: app/src/main/java/com/example/itbook/utils/ActivityExtension.kt [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+package com.example.itbook.utils
+
+import android.app.Activity
+import android.content.Context
+import android.view.View
+import android.view.inputmethod.InputMethodManager
+
+fun Activity.closeKeyboard(view: View) {
+    val imm = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
```
Comment (thonc-2135) [inline app/src/main/java/com/example/itbook/utils/ActivityExtension.kt]: chuyển là View chứ k nên để Activity, thế Fragment lại k dùng đk

File: app/src/main/java/com/example/itbook/ui/favorite/FavoriteBooksFragment.kt [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+        textBooksCount.text = booksCount
+        previewBookAdapter.apply {
+            updateData(books.toMutableList())
+            notifyDataSetChanged()
+        }
+        this.books = books.toMutableList()
+    }
+
+    override fun showError(error: Exception?) {
+        Toast.makeText(activity, error.toString(), Toast.LENGTH_LONG).show()
```
Comment (thonc-2135) [inline app/src/main/java/com/example/itbook/ui/favorite/FavoriteBooksFragment.kt]: tẹo extension funtion cho Toast

File: app/src/main/java/com/example/itbook/ui/favorite/FavoriteBooksFragment.kt (hunk 3 above)
Comment (zGararz) [reply app/src/main/java/com/example/itbook/ui/favorite/FavoriteBooksFragment.kt]: em cho lên đầu rồi ạ

File: app/src/main/java/com/example/itbook/utils/ActivityExtensions.kt [hunk 6]
```diff
@@ -0,0 +1,8 @@
+package com.example.itbook.utils
+
+import android.app.Activity
+import android.widget.Toast
+
+fun Activity.showError(error: String) {
+    Toast.makeText(this, error, Toast.LENGTH_LONG).show()
```
Comment (thonc-2135) [inline app/src/main/java/com/example/itbook/utils/ActivityExtensions.kt]: dùng context

**Review by thonc-2135 — APPROVED** [review]

**Pull request merged**